src/debug.c

```
@@ -579,7 +579,7 @@ NULL
         aofLoadManifestFromDisk();
         aofDelHistoryFiles();
         int ret = loadAppendOnlyFiles(server.aof_manifest);
-        if (ret != AOF_OK && ret != AOF_EMPTY)
+        if (ret != AOF_OK && ret != AOF_EMPTY && ret != AOF_NOT_EXIST)
```

it doesn't seem this was a regression in #9012.
looking at redis 6.2, a missing file would have done `exit` in loadAppendOnlyFile.
an empty file would have returned C_ERR and then replied with an error to the caller.
so it seems the code in #9012 preserved this in a way (other than returning an error).

anyway, history aside, i don't think we want LOADAOF to return with success if the file is missing.
we could either return an error, or exit, considering we already flushed all the data, there's no much difference.
and considering that the test suite that calls this command is likely to be one without `catch`, an error which will be returned will be caught, so i don't really mind if we change the exit to reply with error or not.
but i do think the condition should not include AOF_NOT_EXIST.

---

OK. I tested the behavior of 6.2 and exited directly, let's keep it.